app/uk/gov/hmrc/cdsreimbursementclaimfrontend/controllers/securities/SelectDutiesController.scala

```diff
@@ -16,18 +16,181 @@
+  }
+
+  def show(securityId: String): Action[AnyContent] = actionReadJourney { implicit request => journey =>
+    processAvailableDuties[Result](
+      securityId: String,
+      journey: SecuritiesJourney,
+      request: Request[_],
+      error => {
+        logger.warn(s"No Available duties: $error")
+        Redirect(baseRoutes.IneligibleController.ineligible()).asFuture
+      },
+      dutiesAvailable =>
+        {
+          val emptyForm: Form[DutiesSelectedAnswer] = selectDutiesForm(dutiesAvailable.map(_._1))
+          val filledForm                            = NonEmptyList
+            .fromList(journey.getAllSelectedDuties.map(a => Duty(a._2)).toList)
+            .fold(emptyForm)(emptyForm.fill)
+          Ok(
+            selectDutiesPage(filledForm, securityId, dutiesAvailable, routes.SelectDutiesController.submit(securityId))
+          )
+        }.asFuture
+    )
+  }
+
+  @SuppressWarnings(Array("org.wartremover.warts.Any"))
+  def submit(securityId: String): Action[AnyContent] = actionReadWriteJourney { implicit request => journey =>
+    processAvailableDuties[(SecuritiesJourney, Result)](
+      securityId: String,
+      journey: SecuritiesJourney,
+      request: Request[_],
+      error => {
+        logger.warn(s"No Available duties: $error")
+        (journey, Redirect(baseRoutes.IneligibleController.ineligible())).asFuture
+      },
+      dutiesAvailable => {
+        val form      = selectDutiesForm(dutiesAvailable.map(_._1))
+        val boundForm = form.bindFromRequest()
+        boundForm
+          .fold(
+            errors => {
+              logger.warn(
+                s"Selection of duties to be repaid failed for $securityId because of errors:" +
+                  s"${errors.mkString("", ",", "")}"
+              )
+              (
+                journey,
+                Ok(
+                  selectDutiesPage(
+                    boundForm,
+                    securityId,
+                    dutiesAvailable,
+                    routes.SelectDutiesController.submit(securityId)
+                  )
+                )
+              )
+            },
+            dutiesSelected => {
+              updateAndRedirect(journey, securityId, dutiesSelected)
+            }
+          )
+          .asFuture
+      }
+    )
+  }
+
+  private def updateAndRedirect(
+    journey: SecuritiesJourney,
+    securityId: String,
+    dutiesSelected: DutiesSelectedAnswer
+  ): (SecuritiesJourney, Result) =
+    journey
+      .selectAndReplaceTaxCodeSetForSelectedSecurityDepositId(securityId, dutiesSelected.map(_.taxCode).toList)
+      .fold(
+        error => {
+          logger.warn(error)
+          (journey, Redirect(controllers.routes.IneligibleController.ineligible()))
+        },
+        updatedJourney => {
+          //(updatedJourney, Redirect(routes.EnterClaimController.show(securityId, dutiesSelected.head.taxCode))) todo CDSR-1777
```

Comment:
I think we already have this route implemented as work-in-progress, so you could uncomment.

Reply:
done